optax/_src/transform.py

```diff
@@ -605,3 +605,30 @@ def update_fn(updates, state, params=None):
     return updates, ApplyEvery(count=count_inc % k, grad_acc=grad_acc)
 
   return GradientTransformation(init_fn, update_fn)
+
+
+class CentralState(OptState):
+  """The `centralize` transformation is stateless."""
+
+
+def centralize() -> GradientTransformation:
+  """Centralize gradients.
+
+  References:
+    [Yong et al, 2020](https://arxiv.org/abs/2004.01461)
+
+  Returns:
+    An (init_fn, update_fn) tuple.
+  """
+
+  def init_fn(_):
+    return CentralState()
+
+  def update_fn(updates, state, params=None):
+    del params
+    updates = jax.tree_map(
+        lambda g: g - g.mean(tuple(range(1, len(g.shape))), keepdims=True) if len(g.shape) > 1
+          else g, updates)
```

Comment:
make this a private function

`def _subtract_mean():
    if ...:
       ...
    else:
       ...
```

instead of a lambda.